### PR TITLE
Warn about upcoming default change in `conda_pkg_format`

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

For more information, see https://github.com/conda/conda-build/pull/5527

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](https://github.com/conda/conda-build/blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: conda_build/cli/main_build.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
     get_or_merge_config,
     zstd_compression_level_default,
 )
-from ..utils import LoggingContext
+from ..utils import LoggingContext, get_logger
```
Comment (jezdez) [inline conda_build/cli/main_build.py:27]:
```suggestion
from ..utils import LoggingContext
```

File: conda_build/cli/main_build.py [hunk 2]
```diff
@@ -503,6 +508,18 @@ def parse_args(args: Sequence[str] | None) -> tuple[ArgumentParser, Namespace]:
...
+    if (
+        all(not arg.startswith("--package-format") for arg in args)
+        and warn_about_default_pkg_format
+    ):
+        get_logger(__name__).warning(
+            "The default `pkg_format` value will become '.conda' in 25.1. "
+            "If you want to keep using `.tar.bz2`, consider:\n"
+            "- Setting `conda_build.pkg_format: 'tar.bz2' in your condarc file.\n"
+            "- Using `--pkg-format=tar.bz2` in the CLI.",
+        )
```
Comment (jezdez) [inline conda_build/cli/main_build.py:522]: This also needs an import above from `conda_build.deprecations`:

```suggestion
        deprecated.topic(
            "24.11",
            "25.1",
            topic=f"The default `pkg_format` of '.tar.bz2'",
            addendum=(
                "\n\n"
                "The new default `pkg_format` value will be '.conda'."
                "If you want to keep using `.tar.bz2`, consider:\n"
                "- Setting `conda_build.pkg_format: 'tar.bz2' in your condarc file.\n"
                "- Using `--pkg-format=tar.bz2` in the CLI.\n",
            ),
            deprecation_type=FutureWarning,
        )

```

**Review by jezdez — CHANGES_REQUESTED** [review]
For consistency purposes let's use the deprecations functionality since it's easier to filter for end users than loggers

**Comment by jaimergp** [comment]
Yay, passing now! Ready for review @conda/builds-tools

File: conda_build/cli/main_build.py (hunk 2 above)
Comment (dholth) [reply conda_build/cli/main_build.py]: I use this in my condarc
```
conda-build:
  pkg_format: 2
```

**Review by beeankha — APPROVED** [review]

**Pull request merged**